lib/Dialect/TritonGPU/Transforms/Utility.cpp

```diff
@@ -418,6 +428,18 @@ static Attribute inferDstEncoding(triton::ReshapeOp op, Attribute encoding) {
                                    op.getAllowReorder());
 }
 
+static Attribute inferDstEncoding(GatherOp op, Attribute encoding) {
+  // Don't propagate through if the layout is already optimized.
+  if (op.getEfficientLayout()) {
+    return {};
```

Comment:
Shouldn't we keep "can infer" logic separate from "should propagate"?

Reply:
I'm not actually sure. What are the consequences of allowing inference of layouts through a layout anchor?

Reply:
I agree. Maybe add a special condition in the while loop of `  while (!queue.empty()) {` before `inferSrcEncoding`

Reply:
I think as long as `propagateToUsers` doesn't propagate the encoding to ops marked as `ttg.layout_anchor` then it won't effect the result whether it supports `inferDstEncoding` or not.

Reply:
I'll try this out. Thanks for the tips!

Reply:
@peterbell10 you were right. I removed these checks here